src/EditorFeatures/Core/Classification/Syntactic/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -291,7 +301,7 @@ private async ValueTask ProcessChangesAsync(ImmutableSegmentedList<ITextSnapshot
 
                 lock (_gate)
                 {
-                    _lastProcessedData = (currentSnapshot, currentDocument, currentRoot);
+                    _lastProcessedData = (currentSnapshot, currentRoot is not null ? currentRoot : currentDocument);
```

Comment:
> I'm getting hit hard by this problem locally, but I'm not able to get the fix merged because bootstrap build is failing with this assertion. Can I get some help resolving it?

Looking. The assertion is hit when nullable walker visits this conditional operator. If you want a workaround, I guess you could try changing the code, e.g.:

```suggestion
                    _lastProcessedData = (currentSnapshot, currentRoot ?? currentDocument);
```

Reply:
Ah, this expression is special. I'll see if adding explicit casts helps.

Reply:
I filed #67975 and added the explicit casts, but then I realized I can use a modified version with `??` and switched to that.